hikari/impl/rest.py

```diff
@@ -4609,3 +4610,111 @@ async def edit_stage_instance(
     async def delete_stage_instance(self, channel: snowflakes.SnowflakeishOr[channels_.GuildStageChannel]) -> None:
         route = routes.DELETE_STAGE_INSTANCE.compile(channel=channel)
         await self._request(route)
+
+    async def fetch_auto_mod_rules(
+        self, guild: snowflakes.SnowflakeishOr[guilds.PartialGuild], /
+    ) -> typing.Sequence[auto_mod.AutoModRule]:
+        results = await self._request(routes.GET_GUILD_AUTO_MODERATION_RULES.compile(guild=guild))
+        assert isinstance(results, list)
+        return [self._entity_factory.deserialize_auto_mod_rule(rule) for rule in results]
+
+    async def fetch_auto_mod_rule(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        rule: snowflakes.SnowflakeishOr[auto_mod.AutoModRule],
+        /,
+    ) -> auto_mod.AutoModRule:
+        result = await self._request(routes.GET_GUILD_AUTO_MODERATION_RULE.compile(guild=guild, rule=rule))
+        assert isinstance(result, dict)
+        return self._entity_factory.deserialize_auto_mod_rule(result)
+
+    async def create_auto_mod_rule(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        /,
+        name: str,
+        event_type: typing.Union[int, auto_mod.AutoModEventType],
+        trigger_type: typing.Union[int, auto_mod.AutoModTriggerType],
+        actions: typing.Sequence[auto_mod.PartialAutoModAction],
```

Comment:
I am a bit iffy about using our own models for this.

I think we might want to go down the road of making builders and having the user pass that.

Think it's probably worth moving Embeds to something like that in the future too